engine/execution/computation/computer/computer.go

```diff
@@ -227,11 +227,12 @@ func (e *blockComputer) executeBlock(
 
 	executionData := generateExecutionData(res, collections, systemCol)
 
-	executionDataID, err := e.executionDataProvider.Provide(ctx, block.Height(), executionData)
+	executionDataID, executionDataRoot, err := e.executionDataProvider.Provide(ctx, block.Height(), executionData)
 	if err != nil {
 		return nil, fmt.Errorf("failed to provide execution data: %w", err)
 	}
 
+	res.ExecutionDataRoot = *executionDataRoot
```

Comment:
Can you explain these changes?

It looks like the return type of Provide has been changed, but I don't actually see the change to the Provider in this PR? Also, why did we change from pointer?